modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -0,0 +1,366 @@
+        method: 'POST',
+        mode: 'same-origin',
+        credentials: 'include',
+        redirect: 'follow',
+        headers: {
+          'Accept': 'application/json, text/plain, */*',
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: send,
+      }
+    ).then((response) => response.json())
+      .then(
+        (data) => {
+          // Populate the form errors.
+          if (data.errors && data.errors.length > 0) {
+            this.setState({errors: data.errors});
+          }
+          // Populate the select options for subproject.
+          if (data.subproject) {
+            const state = Object.assign({}, this.state);
+            state.form.options.subproject = data.subproject;
+            state.form.value.subproject = '';
+            state.form.display.subproject = true;
+            this.setState(state);
+          }
+          // Populate the select options for psc.
+          if (data.psc) {
+            const state = Object.assign({}, this.state);
+            state.form.options.psc = data.psc;
+            state.form.value.psc = '';
+            state.form.display.psc = true;
+            this.setState(state);
+          }
+          // Populate the select options for visit.
+          if (data.visit) {
+            // Store the (complete) visit selection information.
+            const state = Object.assign({}, this.state);
+            state.storage.visit = data.visit;
+            this.setState(state);
+            // Handle visit selection.
+            this.handleVisitLabel();
+          }
+          // Display form to user.
+          this.setState({isLoaded: true});
+        }).catch((error) => {
+      this.populateErrors({message: 'Server error.'});
+      this.setState({isLoaded: true});
+    });
+  }
+  /**
+   * Visit Labels refreshes when Subproject changes.
+   */
+  handleVisitLabel() {
+    const state = Object.assign({}, this.state);
+    state.form.options.visit = state.storage.visit[
+      state.form.value.subproject
+      ];
+    if (state.form.value.subproject) {
+      state.form.value.visit = Object.keys(state.storage.visit[
+        state.form.value.subproject
+        ])[0];
+    }
+    state.form.display.visit = true;
+    this.setState(state);
+  }
+  /**
+   * Set the form data based on state values of child elements/components
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setForm(formElement, value) {
+    const state = Object.assign({}, this.state);
+    state.form.value[formElement] = value;
+    this.setState(state);
+    if (formElement === 'subproject') {
+      this.handleVisitLabel();
+    }
+  }
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    const state = Object.assign({}, this.state);
+    state.form.options.required = {
+      subproject: true,
```

Comment:
i think this should be done in render() and not handleSubmit() because a post request is still being sent here even if fields are empty.
<img width="1276" alt="Screenshot 2019-04-01 14 32 20" src="https://user-images.githubusercontent.com/17415878/55351289-e921bb00-548b-11e9-8c28-9ea617537b02.png">

I think a better solution to requiring fields is to set their required prop to true but turn off the error message. the html form can then throw the required error on the browser itself, i.e. ..

<img width="1278" alt="Screenshot 2019-04-01 14 43 14" src="https://user-images.githubusercontent.com/17415878/55351537-9bf21900-548c-11e9-9faf-8ceaee52a078.png">

Reply:
Okay i'll fix this tomorrow, have to go but note for me from @zaliqarosli is to set the values to null and not empty string.